pkg/finder/podresourcefinder.go

```diff
@@ -0,0 +1,147 @@
+	if finderInstance.args.Namespace != "" {
+		log.Printf("watching namespace %q", finderInstance.args.Namespace)
+	} else {
+		log.Printf("watching all namespaces")
+	}
+
+	return finderInstance, nil
+}
+
+func (f *PodResourceFinder) GetNodeResourceData() *NodeResources {
+	return f.NodeResourceData
+}
+
+func (f *PodResourceFinder) isWatchable(podNamespace string) bool {
+	if f.args.Namespace == "" {
+		return true
+	}
+	//TODO:  add an explicit check for guaranteed pods
+	return f.args.Namespace == podNamespace
+}
+
+func (f *PodResourceFinder) Scan() ([]PodResources, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultPodResourcesTimeout)
+	defer cancel()
+
+	//Pod Resource API client
+	resp, err := f.podResourceClient.List(ctx, &podresourcesapi.ListPodResourcesRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("Can't receive response: %v.Get(_) = _, %v", f.podResourceClient, err)
+	}
+
+	var podResData []PodResources
+
+	for _, podResource := range resp.GetPodResources() {
+		if !f.isWatchable(podResource.GetNamespace()) {
+			log.Printf("SKIP pod %q\n", podResource.Name)
+			continue
+		}
+
+		podRes := PodResources{
+			Name:      podResource.GetName(),
+			Namespace: podResource.GetNamespace(),
+		}
+		for _, container := range podResource.GetContainers() {
+			contRes := ContainerResources{
+				Name: container.Name,
+			}
+			devs := make(map[string][]string)
+			for _, device := range container.GetDevices() {
+				devs[device.ResourceName] = device.DeviceIds
+			}
+			cpuList := container.GetCpuIds()
+			contRes.Resources = append(contRes.Resources, makeCPUResourceInfo(cpuList)...)
+			contRes.Resources = append(contRes.Resources, makePCIDeviceResourceInfo(devs, f.NodeResourceData.pci2ResourceMap)...)
+			log.Printf("pod %q container %q contData=%s\n", podResource.GetName(), container.Name, spew.Sdump(contRes))
+			podRes.Containers = append(podRes.Containers, contRes)
+		}
+
+		podResData = append(podResData, podRes)
+
+	}
+
+	return podResData, nil
+}
+
+func makeCPUResourceInfo(cpus []uint32) []ResourceInfo {
+	var ret []string
+	for _, cpuID := range cpus {
+		ret = append(ret, fmt.Sprintf("%d", cpuID))
+	}
+
+	fmt.Printf("makeCPUResourceInfo: data %v", ret)
+	return []ResourceInfo{
+		{
+			Name: v1.ResourceCPU,
+			Data: ret,
+		},
+	}
+}
+
+func makePCIDeviceResourceInfo(devs map[string][]string, pci2ResMap map[string]string) []ResourceInfo {
+	var resInfos []ResourceInfo
+	//slice of deviceIDs (PCI addresses)
+	pciAddrs := make([]string, 0)
+	var resName string
+	var ok bool
+	for _, devIds := range devs {
+		for _, devId := range devIds {
+			resName, ok = pci2ResMap[devId]
+			if !ok {
+				continue
+			}
+			pciAddrs = append(pciAddrs, devId)
+		}
+
+	}
+
+	fmt.Printf("makePCIDeviceResourceInfo: data %v", pciAddrs)
+	resInfos = append(resInfos, ResourceInfo{
+		Name: v1.ResourceName(resName),
+		Data: pciAddrs,
+	})
+	return resInfos
+}
```

Comment:
it seems to me that we have duplicated code. Can be fixed later.

Reply:
So, the code seems duplicate but if you look carefully there are differences because of difference in the format in which data is obtained from CRI and podresource API. 

- For cpu information, CRI returns CPUSet whereas PodResource api returns []unit32

- For device information, in CRI we have a separate comma separated device PCI addresses whereas in case of pod resource api we get a slice of devices.

But, I certainly see too this can be improved by finding least common denominator and/or a bit of type conversion. Will address this in a separate PR later